schematic/schemas/data_model_validator.py

```diff
@@ -143,72 +134,190 @@ def check_blacklisted_characters(self) -> list[str]:
             list[str]: list of warnings for each node in the graph, that has a Display
               name that contains blacklisted characters.
         """
-        warning = []
-        for _, node_dict in self.graph.nodes(data=True):
-            if any(
-                bl_char in node_dict["displayName"]
-                for bl_char in self.blacklisted_chars
-            ):
-                node_display_name = node_dict["displayName"]
-                blacklisted_characters_found = [
-                    bl_char
-                    for bl_char in self.blacklisted_chars
-                    if bl_char in node_dict["displayName"]
-                ]
-                blacklisted_characters_str = ",".join(blacklisted_characters_found)
-                warning.append(
-                    (
-                        f"Node: {node_display_name} contains a blacklisted character(s): "
-                        f"{blacklisted_characters_str}, they will be striped if used in "
-                        "Synapse annotations."
-                    )
-                )
-        return warning
+        return check_characters_in_node_display_name(
+            self.graph.nodes(data=True), BLACKLISTED_CHARACTERS
+        )
 
-    def check_reserved_names(self) -> list[str]:
+    def _check_reserved_names(self) -> list[str]:
         """Identify if any names nodes in the data model graph are the same as reserved name.
         Returns:
-            error, list: List of erros for every node in the graph whose name overlaps
+            error, list: List of errors for every node in the graph whose name overlaps
               with the reserved names.
         """
-        error = []
-        reserved_names_found = [
-            (name, node)
-            for node in self.graph.nodes
-            for name in self.reserved_names
-            if name.lower() == node.lower()
+        reserved_names_found = match_node_names_with_reserved_names(
+            self.graph.nodes, RESERVED_NAMES
+        )
+        return create_reserve_name_error_messages(reserved_names_found)
+
+
+def get_node_labels_from(input_dict: dict) -> list:
```

Comment:
Is there a more specific return typing than list? Maybe

```
from typing into List

def get_node_labels_from(input_dict: dict) -> List[str]:
```

Reply:
I don't think so, for the same reason the input dict is not more specific. The dict is from networkx.Multidigraph.nodes. And it looks like the values can be anything, and the nested dicts cna be anything as well. In the context of Schematic, the "node_label" field has so far always been a string, but I don't think we should gauruntee it in the return type.